game_event/tests.py

```diff
@@ -252,3 +273,47 @@ def test_fail_add_player_to_event_already_in_it(self, client, game_event, player
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestGetWeatherFromApi:
+    def test_one_valid_response(self, game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+        output = get_weather_from_api([game_event], mock_urlopen)
+        expected_weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        assert output == [expected_weather]
+
+    def test_exception_thrown_from_api_call(self, game_event, player, mock_urlopen):
+
+        mock_urlopen.side_effect = Exception()
+
+        output = get_weather_from_api([game_event], mock_urlopen)
+        assert output[0] is None
```

Comment:
```suggestion
        assert output == [None]
```

Reply:
We get back from the `get_weather_from_api` a list of game_events. from what I understand in that case the weather field will be `None `. Is `output == [None]` assertion checking that? 
`output[0].weather`  is the way to access the weather no? @yftacherzog

Reply:
The comment was written when we were still returning just the weathers.
We know that we're limited in our ability to compare against the entire object, but at least I'd like us to get some notion that the list is containing the stuff that we expect and only that.

With that in mind, you can do something like:
```python
weathers = [e.weather for e in output]
assert weathers = [None]
```
It's not perfect, but it's verifying that we're getting the expected number of elements.

You can do something similar also when you have more than one instance.